### PR TITLE
Fix legacy pagination for Subscribed feed

### PR DESCRIPTION
**Pull request by maxpoulin64**
This is the fix suggested in #4019

I've manually tested both legacy pagination (`page=2&limit=20`) as well as the new pagination (`page_cursor=XXXXXXX&limit=20`) using the same endpoint as #4019, and both methods appear to work as expected.

### CODE REVIEW

**Comment by phiresky** [comment]
Ah. In a previous version of the code, the whole function `prefetch_upper_bound_for_page_before` was only called in the case that page_after was set.

So now the question is does it make sense to call the estimator with legacy pagination? It seems like the page/offset is passed through to the estimation query, and then later also to the real query. So a user requesting page=3 _should_ be the same as the user requesting page 1 with limit = 3x(normal limit) and filtering out the first two pages of results. I'm not sure though, the interaction of these parameters is kind of complex to understand even though I wrote it.

This would mean that the estimator should still be helpful for performance of the Subscribed view even with legacy pagination, as long as page numbers are fairly low (maybe < 10% of pages so if page number < 10 and total page count would be 100). If you set page=5&limit=30, with estimator it should be fetching 2x 5x30 posts, while without estimator it would (worst case = subscribed to only unpopular communities) be fetching  all posts.


---
tldr: I'm not 100% sure if the results will be correct with this, but it should be a correct solution. If the results turn out to be wrong, this would be an alternative patch with worse perf for legacy pagination:

```diff
diff --git a/crates/db_views/src/post_view.rs b/crates/db_views/src/post_view.rs
index 87c31d7e5..99a2ad598 100644
--- a/crates/db_views/src/post_view.rs
+++ b/crates/db_views/src/post_view.rs
@@ -684,6 +684,7 @@ impl<'a> PostQuery<'a> {
       && self.community_id.is_none()
       && self.local_user.is_some()
       && self.page_before_or_equal.is_none()
+      && self.page.is_none()
     {
       if let Some(query) = self.prefetch_upper_bound_for_page_before(pool).await? {
         queries().list(pool, query).await
```

**Review by Nutomic — APPROVED** [review]

**Comment by Nutomic** [comment]
@phiresky Sounds like we can merge it like this and if another issue is reported we need to revisit it.
